deck/views.py

```diff
@@ -31,16 +31,36 @@ class BaseEventView(object):
 class ListEvents(BaseEventView, ListView):
     template_name = 'event/event_list.html'
     queryset = Event.objects.published_ones()
+    allow_empty = True
 
     def get_context_data(self, **kwargs):
         context = super(ListEvents, self).get_context_data(**kwargs)
         if self.request.user.is_authenticated():
-            event_list = Event.objects.filter(
-                models.Q(is_published=True) |
+            event_list = context['event_list'].filter(
                 models.Q(author=self.request.user))
             context.update(event_list=event_list)
         return context
 
+    def post(self, request, *args, **kwargs):
```

Comment:
POST is not the best HTTP method to allow users to search for something.
The first argument is: 
* GET allows cache. POST must not be cached.
* After a POST it is harder to keep some browser state like what you used to query the results.
* Using POST is not the best approach because users will not be able to make filters using URL arguments.